clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -6148,6 +6148,35 @@ NamedDecl *Sema::FindInstantiatedDecl(SourceLocation Loc, NamedDecl *D,
   return D;
 }
 
+static void processSYCLKernel(Sema &S, FunctionDecl *FD, MangleContext &MC) {
+  // Because all SYCL kernel functions are template functions - they
+  // have deferred instantination. We need bodies of these functions
+  // so we are checking for SYCL kernel attribute after instantiation.
+  if (S.LangOpts.SYCLIsDevice) {
+    S.ConstructOpenCLKernel(FD, MC);
+  } else if (S.LangOpts.SYCLIsHost) {
+    auto CRD = (*FD->param_begin())->getType()->getAsCXXRecordDecl();
```

Comment:
```suggestion
    auto *CRD = (*FD->param_begin())->getType()->getAsCXXRecordDecl();
```
However, I would prefer explicit type here.

Reply:
Used explicit type here.